libs/apigateway-lambda-proxy/src/lib/metric-alarms.ts

```diff
@@ -0,0 +1,325 @@
+        )
+
+        new ErrorRate4xxAlarm(
+            name,
+            {
+                snsTopicArn: args.snsTopicArn,
+                apiGateway: args.apiGateway,
+                getTags: args.getTags,
+                errorRate4xxPercent: args.thresholds?.errorRate4xxPercent,
+            },
+            { parent: this },
+        )
+
+        new IntegrationLatencyAlarm(
+            name,
+            {
+                snsTopicArn: args.snsTopicArn,
+                apiGateway: args.apiGateway,
+                getTags: args.getTags,
+                stdDeviations: args.thresholds?.integrationLatencyStdDeviations,
+            },
+            { parent: this },
+        )
+    }
+}
+
+/**
+ * Get dimensions for a cloudwatch alarm
+ *
+ * @param api the APIGateway to test
+ * @returns the dimensions for passing to CloudWatch alarms
+ */
+function getDimensions(api: HttpGateway) {
+    return pulumi.all([api.id, api.stage]).apply(([id, stage]) => ({
+        ApiId: id,
+        ...(stage
+            ? {
+                  Stage: stage,
+              }
+            : {}),
+    }))
+}
+
+/** raise an alarm if the 5xx error rate exceeds errorRate5xxPercent */
+export class ErrorRate5xxAlarm extends pulumi.ComponentResource {
+    constructor(
+        name: string,
+        args: {
+            /**
+             * Alerts will be sent to this SNS topic
+             */
+            snsTopicArn: pulumi.Input<string>
+            /**
+             * The API Gateway to monitor
+             */
+            apiGateway: HttpGateway
+            /**
+             * Alert if the 5xx error rate exceeds this
+             * (default: 2 percent)
+             */
+            errorRate5xxPercent?: pulumi.Input<number>
+            /**
+             * a callback function that returns tags for
+             * each resource
+             */
+            getTags: (
+                name: string,
+            ) => {
+                [key: string]: pulumi.Input<string>
+            }
+        },
+        opts?: pulumi.ComponentResourceOptions,
+    ) {
+        super('wanews:apigateway/ErrorRate5xxAlarm', name, {}, opts)
+
+        const metricName = '5xx'
+        const dimensions = getDimensions(args.apiGateway)
+        const thresholdPercent = args.errorRate5xxPercent ?? 2
+        const threshold = pulumi
+            .output(thresholdPercent)
+            .apply((percent) => percent / 100)
+        const resourceName = `${name}-5xx-rate`
+
+        new aws.cloudwatch.MetricAlarm(
+            resourceName,
+            {
+                evaluationPeriods: 2,
+                datapointsToAlarm: 2,
+                comparisonOperator: 'GreaterThanThreshold',
+                namespace,
+                metricName,
+                dimensions,
+                threshold,
+                statistic: 'Average',
+                alarmDescription: pulumi.interpolate`threshold: 5xx rate >= ${thresholdPercent}%: ${args.apiGateway.id}`,
```

Comment:
Description not updated :)

Reply:
<img width="780" alt="image" src="https://user-images.githubusercontent.com/1610227/121999510-fcf13780-cddf-11eb-810c-3fee45aa7de0.png">

It looks up to date? 😕

Could you please be specific? What are you suggesting instead? Is there some error?

Reply:
Ah, `GreaterThan` should be `>` not `>=`

Reply:
This will be fixed in #31